ts/Core/Axis/Color/ColorAxis.ts

```diff
@@ -564,21 +565,38 @@ class ColorAxis extends Axis implements AxisLike {
         this.setLegendColor();
 
         // Create the gradient
-        item.legendSymbol = this.chart.renderer.rect(
-            0,
-            (legend.baseline as any) - 11,
-            width,
-            height
-        ).attr({
-            zIndex: 1
-        }).add(item.legendGroup);
-
+        if (!item.legendSymbol) {
+            item.legendSymbol = this.chart.renderer.rect(
+                0,
+                (legend.baseline as any) - 11,
+                width,
+                height
+            ).attr({
+                zIndex: 1
+            }).add(item.legendGroup);
+        }
+        const options = legend.options,
+            symbolPadding = options.symbolPadding || 0,
+            series = (item as any).series.drawLegendSymbol ?
+                (item as any).series :
+                item,
+            seriesOptions = series.options,
+            showCheckbox = legend.createCheckboxForItem &&
+                seriesOptions &&
+                seriesOptions.showCheckbox,
+            itemExtraWidth = legend.symbolWidth + symbolPadding +
+                        itemDistance + (showCheckbox ? 20 : 0);
         // Set how much space this legend item takes up
         axis.legendItemWidth = (
             width +
             padding +
-            (horiz ? itemDistance : labelPadding)
-        );
+            (horiz ? itemDistance : labelPadding) + (
+                (options.layout === 'vertical') &&
+                (item as ColorAxis).maxLabelLength &&
+                ((item as ColorAxis).maxLabelLength > itemExtraWidth) &&
+                ((item as ColorAxis).maxLabelLength - itemExtraWidth) ||
```

Comment:
Those castings don't look like needed - `item` has its type defined as the function argument.